include/RevCore.h

```diff
@@ -424,6 +424,7 @@ class RevCore {
   EcallStatus ECALL_fchownat();               // 54, rev_fchownat(int dfd, const char  *filename, uid_t user, gid_t group, int flag)
   EcallStatus ECALL_fchown();                 // 55, rev_fchown(unsigned int fd, uid_t user, gid_t group)
   EcallStatus ECALL_openat();                 // 56, rev_openat(int dfd, const char  *filename, int flags, umode_t mode)
+  EcallStatus ECALL_open();                   // 1024, rev_open(const char  *filename, int flags)
```

Comment:
I think these are in order based on the ecall code so I would move this down to probably here? 

![image](https://github.com/user-attachments/assets/25206ec5-077a-49e0-85ed-d4d22d7e63fa)

Maybe @leekillough can decide where it makes more sense. I arbitrarily added 1000 as the pthread functions way back so unfortunately this one having the code 1024 doesn't fit in nicely with the ordering. Really just a nit more than anything

Reply:
The support was a modified copy+paste of the `openat` implementation. Wasn't sure if keeping similar implementations closer together makes sense or to retain the current numerical ordering. Either way, small enough fix.

Reply:
Yeah I could be convinced either way. Will leave it to @leekillough to make the call

Reply:
I would move it so that it is in numerical order with other system calls.

Is the value `1024` special? I thought most system calls were < `512` and that we added "custom" system calls >= `1024` whose number was not tied to any kernel system call but was our own "pseudo-system-call".

Also a number like `1024`, being right on a power-of-2 boundary, might conflict with other system call numbers, so if we are choosing the value to be something special, we should choose a larger value and put it in the next multiple-of-100 after what we have already, or something like that.

Reply:
Cool, I'll move it soon!

Yep! 1024 is the syscall table identifier for `open`. I've some kernels that break when the `open` system call is made and the assembly has the ecall code set to 1024.

I'll try to dig up the value in the kernel this link is what I could dig up quickly.

https://github.com/westerndigitalcorporation/RISC-V-Linux/blob/master/riscv-pk/pk/syscall.h

Reply:
@leekillough looks like we'll have to do a cross compile to risc-v assembly and double check. the gcc risc-v backend doesn't define a system call number for `open`; getting the impression `open` is implemented _over_
`openat`.

https://github.com/riscv-collab/riscv-gnu-toolchain/blob/master/linux-headers/include/asm-generic/unistd.h